### PR TITLE
Use OpenSSL md5 if it is available

### LINKED ISSUE #1582: Use system md5
Like https://sources.debian.org/patches/ddnet/11.8-1/use-system-libmd.patch/

### PR DESCRIPTION
**Pull request by heinrich5991**
Fixes #1582.

### CODE REVIEW

File: src/engine/client/serverbrowser.cpp [hunk 1]
```diff
@@ -92,11 +92,11 @@ const CServerInfo *CServerBrowser::SortedGet(int Index) const
...
-	MD5_CTX Md5;
-	md5_init(&Md5);
-	md5_update(&Md5, m_aTokenSeed, sizeof(m_aTokenSeed));
-	md5_update(&Md5, (unsigned char *)&Addr, sizeof(Addr));
-	MD5_DIGEST Digest = md5_finish(&Md5);
+	SHA256_CTX Sha256;
+	sha256_init(&Sha256);
+	sha256_update(&Sha256, m_aTokenSeed, sizeof(m_aTokenSeed));
+	sha256_update(&Sha256, (unsigned char *)&Addr, sizeof(Addr));
+	SHA256_DIGEST Digest = sha256_finish(&Sha256);
```
Comment (def-) [inline src/engine/client/serverbrowser.cpp]: Isn't SHA256 much slower than md5?

**Comment by def-** [comment]
LGTM

File: src/engine/client/serverbrowser.cpp (hunk 1 above)
Comment (def-) [reply src/engine/client/serverbrowser.cpp]: And is this token colliding much with md5?

Comment (heinrich5991) [reply src/engine/client/serverbrowser.cpp]: SHA256 vs MD5 shouldn't matter much and MD5 is discouraged, so I thought we should maybe just use SHA256 by default.

I can revert this if you don't like it.

Comment (def-) [reply src/engine/client/serverbrowser.cpp]: I'm just afraid of making things a bit slower and slower, adds up over time. But it's ok

**Pull request merged**